Mstp yang

wajahatrazi opened the pull request.

wajahatrazi committed:
Adding STP YANG model

wajahatrazi committed:
Adding STP YANG model

wajahatrazi committed:
Adding STP YANG model

wajahatrazi committed:
Adding STP YANG model

divyachandralekha commented on src/sonic-yang-models/yang-models/sonic-stp.yang:
Please include pvst fileds also.

wajahatrazi replied:
Hi Divya, I have added the changes for both pvst & mstp.

wajahatrazi replied:
Please review, I have updated

wajahatrazi committed:
Updating the YANG model for both PVST & MST

wajahatrazi committed:
Updating the YANG model for both PVST & MST

divyachandralekha commented on src/sonic-yang-models/yang-models/sonic-stp.yang:
```diff
@@ -0,0 +1,236 @@
+                default "30";
+                description "Root guard timeout in seconds";
+            }
+            leaf forward_delay {
```
grouping vlanModeAttr {
        leaf forward_delay {
            type uint8 {
                range "4..30" {
                    error-message "Invalid Forwarding Delay value.";
                }
            }
            units seconds;
            default 15;
            description
                "The delay used by STP bridges to transition root and
                designated ports to forwarding";
        }

        leaf hello_time {
            type uint8 {
                range "1..10" {
                    error-message "Invalid Hello Time value.";
                }
            }
            units seconds;
            default 2;
            description
                "The interval between periodic transmissions of
                configuration messages by designated ports";
        }

        leaf max_age {
            type uint8 {
                range "6..40" {
                    error-message "Invalid Maximum Age Time value.";
                }
            }
            units seconds;
            default 20;
            description
                "The maximum age of the information transmitted by the
                bridge when it is the root bridge";
        }

        leaf priority {
            type uint16 {
                range "0..61440" {
                    error-message "Invalid Bridge Priority value.";
                }
            }
            sonic-ext:custom-validation ValidateBridgePriority;
            default 32768;
            description
                "The manageable component of the Bridge Identifier";
        }
    }

Add this , make add as below on this block
  uses vlanModeAttr;

wajahatrazi replied:
Please review, I have updated

divyachandralekha commented on src/sonic-yang-models/yang-models/sonic-stp.yang:
```diff
@@ -0,0 +1,236 @@
+                default "32768";
+                description "Bridge priority";
+            }
+            
```
Close the bracket here. MST block should be as below
      container STP_MST {
            list STP_MST_LIST {
                max-elements 1;
                key "keyleaf";
                sonic-ext:dependent-on "STP_LIST";

                leaf keyleaf {
                    type enumeration {
                        enum GLOBAL;
                    }
                    description
                        "Key node identifier. It's value is always GLOBAL";
                }

                leaf name {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type string;
                    description
                        "MST Region name";
                }

                leaf revision {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint32;
                    description
                        "MST Revision number";
                }

                leaf max_hops {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST Max hops";
                }

                leaf hello_time {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST hello time";
                }

                leaf max_age {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST max age";
                }

                leaf forward_delay {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST forward delay";
                }

                leaf hold_count {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST hold count";
                }
            }
        }

wajahatrazi replied:
Please review, I have updated

divyachandralekha replied:
module sonic-stp {
    namespace "http://github.com/sonic-net/sonic-stp";
    prefix stp;
    yang-version 1.1;

    import sonic-extension {
        prefix sonic-ext;
    }

    description "SONiC STP YANG model for PVST and MST configurations";

    revision 2024-11-24 {
        description "Version01 for combined PVST and MST configurations.";
    }

    grouping vlanModeAttr {
        leaf forward_delay {
            type uint8 {
                range "4..30" {
                    error-message "Invalid Forwarding Delay value.";
                }
            }
            units seconds;
            default 15;
            description
                "The delay used by STP bridges to transition root and
                designated ports to forwarding";
        }

        leaf hello_time {
            type uint8 {
                range "1..10" {
                    error-message "Invalid Hello Time value.";
                }
            }
            units seconds;
            default 2;
            description
                "The interval between periodic transmissions of
                configuration messages by designated ports";
        }

        leaf max_age {
            type uint8 {
                range "6..40" {
                    error-message "Invalid Maximum Age Time value.";
                }
            }
            units seconds;
            default 20;
            description
                "The maximum age of the information transmitted by the
                bridge when it is the root bridge";
        }

        leaf priority {
            type uint16 {
                range "0..61440" {
                    error-message "Invalid Bridge Priority value.";
                }
            }
            default 32768;
            description
                "The manageable component of the Bridge Identifier";
        }
    }

    grouping interfaceAttr {
        leaf path_cost {
            type uint64 {
                range "1..200000000" {
                    error-message "Invalid Port Path Cost value.";
                }
            }
            default 200;
            description
                "The port's contribution, when it is the Root Port,
                to the Root Path Cost for the Bridge";
        }

        leaf priority {
            type uint8 {
                range "0..240" {
                    error-message "Invalid Port Priority value.";
                }
            }
            default 128;
            description
                "The manageable component of the Port Identifier,
                also known as the Port Priority";
        }
    }


    container sonic-spanning-tree {

        container STP {
            list STP_LIST {
                max-elements 1;
                key "keyleaf";
                leaf keyleaf {
                    type enumeration {
                        enum GLOBAL;
                    }
                    description
                        "Key node identifier. It's value is always GLOBAL";
                }

                leaf mode {
                    type enumeration {
                        enum pvst;
                        enum mst;
                    }
                    mandatory true;
                    description
                        "Spanning tree mode";
                }

                leaf rootguard_timeout {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode!='mst'" {
                        error-message "Configuration not allowed in MST mode";
                        error-app-tag stp-invalid;
                    }

                    type uint16 {
                        range "5..600" {
                            error-message "Invalid Root-guard Timeout value.";
                        }
                    }
                    units seconds;
                    description
                        "Once superior BPDUs stop coming on the port, device
                        will wait for a period until root guard timeout before
                        moving the port to forwarding state";
                }

                uses vlanModeAttr;
            }
        }

        container STP_VLAN {
            list STP_VLAN_LIST {
                key "name";
                leaf name {
                    type string;
                    description
                        "Vlan identifier";
                }

                leaf vlanid {
                    type uint16 {
                        range "1..4095" {
                            error-message "Vlan ID out of range";
                            error-app-tag vlanid-invalid;
                        }
                    }
                    description
                        "Vlan identifier number";
                }

                leaf enabled {
                    type boolean;
                    mandatory true;
                    description
                        "Spanning tree enabled/disabled on Vlan";
                }

                uses vlanModeAttr;
            }
        }

        container STP_VLAN_PORT {
            list STP_VLAN_PORT_LIST {
                key "vlan-name ifname";

                leaf vlan-name {
                    type leafref {
                        path "../../../STP_VLAN/STP_VLAN_LIST/name";
                    }
                    description
                        "Reference to Vlan";
                }

                leaf ifname {
                    type leafref {
                        path "../../../STP_PORT/STP_PORT_LIST/ifname";
                    }
                    description
                        "Reference to Ethernet interface or PortChannel";
                }

                uses interfaceAttr;
            }
        }

        container STP_PORT {
            list STP_PORT_LIST {
                key "ifname";
                sonic-ext:dependent-on "STP_LIST";
                leaf ifname {
                    type string;
                    description
                        "Reference to Ethernet interface or PortChannel";
                }

                leaf enabled {
                    type boolean;
                    mandatory true;
                    description
                        "Spanning tree enabled/disabled on Interface";
                }

                leaf root_guard {
                    type boolean;
                    description
                        "Enable/Disable Root guard on port";
                }

                leaf bpdu_guard {
                    type boolean;
                    description
                        "Enable/Disable port BPDU guard";
                }

                leaf bpdu_guard_do_disable {
                    type boolean;
                    description
                        "Port to be disabled when it receives a BPDU";
                }

                leaf uplink_fast {
                    type boolean;
                    description
                        "Enable/Disable uplink-fast on port";
                }

                leaf portfast {
                    must "current()!='true' or ../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='pvst'" {
                        error-message "Configuration not allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type boolean;
                    description
                        "Enable/Disable portfast on port";
                }

                uses interfaceAttr;

                // For MST
                leaf edge_port {
                    type boolean;
                    description
                        "Enable/Disable Edge-port on interface";
                }

                leaf link_type {
                    type enumeration {
                        enum auto;
                        enum shared;
                        enum point-to-point;
                    }
                    description
                        "Specifies the interface's link type. Permissible values
                        are 'shared', 'point-to-point' and 'auto'";
                }
            }
        }

        container STP_MST {
            list STP_MST_LIST {
                max-elements 1;
                key "keyleaf";
                sonic-ext:dependent-on "STP_LIST";

                leaf keyleaf {
                    type enumeration {
                        enum GLOBAL;
                    }
                    description
                        "Key node identifier. It's value is always GLOBAL";
                }

                leaf name {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type string;
                    description
                        "MST Region name";
                }

                leaf revision {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint32;
                    description
                        "MST Revision number";
                }

                leaf max_hops {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST Max hops";
                }

                leaf hello_time {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST hello time";
                }

                leaf max_age {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST max age";
                }

                leaf forward_delay {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST forward delay";
                }

                leaf hold_count {
                    must "../../../STP/STP_LIST[keyleaf='GLOBAL']/mode='mst'" {
                        error-message "Configuration allowed in MST mode";
                        error-app-tag stp-invalid;
                    }
                    type uint8;
                    description
                        "MST hold count";
                }
            }
        }
        container STP_MST_INST {
            list STP_MST_INST_LIST {
                key "instance";

                leaf instance {
                    type uint16;
                    description
                        "Instance identifier";
                }

                leaf-list vlan {
                    type string;
                    description
                        "Vlan list";
                }

                leaf bridge_priority {
                    type uint16 {
                        range "0..61440" {
                            error-message "Invalid Bridge Priority value.";
                        }
                    }
                    description
                        "The manageable component of the Bridge Identifier";
                }
            }
        }

        container STP_MST_PORT {
            list STP_MST_PORT_LIST {
                key "inst_id ifname";

                leaf inst_id {
                    type leafref {
                        path "../../../STP_MST_INST/STP_MST_INST_LIST/instance";
                    }
                    description
                        "Reference to MST Instance";
                }

                leaf ifname {
                    type leafref {
                        path "../../../STP_PORT/STP_PORT_LIST/ifname";
                    }

                    description
                        "Reference to Ethernet interface or PortChannel";
                }
                uses interfaceAttr;
            }
        }
    }  
}

divyachandralekha commented on src/sonic-yang-models/yang-models/sonic-stp.yang:
```diff
@@ -0,0 +1,236 @@
+            }
+        }
+
+        list vlan {
```
Vlan block should be like this 

 container STP_VLAN {
            list STP_VLAN_LIST {
                key "name";
                sonic-ext:custom-validation ValidateStpMaxInstance;
                leaf name {
                    type string;
                    description
                        "Vlan identifier";
                }

                leaf vlanid {
                    type uint16 {
                        range "1..4095" {
                            error-message "Vlan ID out of range";
                            error-app-tag vlanid-invalid;
                        }
                    }
                    description
                        "Vlan identifier number";
                }

                leaf enabled {
                    type boolean;
                    sonic-ext:custom-validation ValidateStpDisableVlan;
                    mandatory true;
                    description
                        "Spanning tree enabled/disabled on Vlan";
                }

                uses vlanModeAttr;
            }
        }

wajahatrazi replied:
Please review, I have updated

divyachandralekha commented on src/sonic-yang-models/yang-models/sonic-stp.yang:
```diff
@@ -0,0 +1,236 @@
+            }
+        }
+
+        list interface {
```
This should be next container

wajahatrazi replied:
Please review, I have updated

divyachandralekha commented on src/sonic-yang-models/yang-models/sonic-stp.yang:
```diff
@@ -0,0 +1,236 @@
+            }
+        }
+
+        list mst_port {
```
this should be next container

wajahatrazi replied:
Please review, I have updated

divyachandralekha reviewed (commented):
Lets meet and discuss

wajahatrazi committed:
I have done changes in the YANG for MSTP

wajahatrazi committed:
I have done changes in the YANG for MSTP

wajahatrazi committed:
Merge branch 'master' into mstp-yang

wajahatrazi committed:
Updating testcases for MSTP YANG

wajahatrazi committed:
Updating testcases for MSTP YANG

volodymyrsamotiy committed:
[Mellanox] Update SDK/FW to 4.7.2110/2014.2110 (#21024)

Signed-off-by: Volodymyr Samotiy <volodymyrs@nvidia.com>

patrickmacarthur committed:
[broadcom-dnx]: Install rdb-cli on syncd-dnx (#20948)

This was missed on the original PR and caused show_techsupport sonic-mgmt tests to fail.

Why I did it
PR #19268 replaced redis-rdb-tool with rdb-cli in the Broadcom XGS syncd container. However, it did not do the same for the DNX syncd container, leading to sonic-mgmt failures in show_techsupport tests on DNX platforms.

How I did it
I applied the same change to the Broadcom DNX syncd container.

How to verify it
Run sonic-mgmt show_techsupport tests on a Broadcom DNX DUT.

mlok-nokia committed:
[Nokia]{asic-sensors] Modify the platform.json to enable the asic_sensors polling on Nokia Platforms (#20979)

Enable and use the asic_sensors polling on the Nokia platforms:
arm64-nokia_ixs7215_52xb-r0
x86_64-nokia_ixr7220_h4-r0
x86_64-nokia_ixr7220_h4_32d-r0
x86_64-nokia_ixr7220_h5_64d-r0
Fixes Nokia-ION/ndk#36

Signed-off-by: mlok <marty.lok@nokia.com>

stepanblyschak committed:
[FC] remove FC delay status field (#20495)

Related to sonic-net/sonic-swss#3326.

Why I did it
Simplify approach to delaying counters on warm boot and fast boot. Removed FLEX_COUNTER_DELAY_STATUS_FIELD and instead postpone all FC processing to happen after apply view to not delay data plane configuration.

The CONFIG_DB should not be updated in runtime anymore for counters to be delayed.

To address #20302.

Work item tracking
Microsoft ADO (number only):
How I did it
Removed FLEX_COUNTER_DELAY_STATUS_FIELD set in enable_counters.py.

How to verify it
Run warm-boot - make sure FC orch runs only after APPLY_VIEW.

saiarcot895 committed:
[database]: Add `--protected-mode no` flag to redis-chassis (#21010)

Redis-chassis doesn't bind to the localhost address intentionally.
Because of this, Redis blocks TCP connections that are connecting to
non-loopback interface addresses when there's no password set (which is
the case for us) unless `--protected-mode no` is set. This has already
been done for docker-database, but now needs to be done for
docker-sonic-vs following the upgrade to Bookworm.

Therefore, in the supervisord.conf file, add `--protected-mode no` to
the arguments passed in for redis-chassis.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>

vincentpcng committed:
Platform DCS-7060DX5-64S misc update (#20323)

Why I did it
This is to add the HWSKU DCS-7060DX5-64S misc update. This is a must have for the platform support. This should get into the master branch as soon as possible and back port to the needed releases.

Work item tracking
Microsoft ADO (number only):
How I did it
Adding additional media, port config settings etc needed for the HWSKU.
This is an updated version of an older PR #19109 for 202305 which is closed. This PR should replace the older PR#19109 and should be cast to all active releases. This PR has the latest chip configuration base on Arista lab setting and the most up to date SI setting for optical transceivers.

How to verify it
We have verified it in house using the sonic mgmt test.
In additional, we have verify all the SI value which match our HW database for optical deployment.
This changes has been verified in Arista lab with 202305 , 202311, 202405 releases.

bobbymcgonigle committed:
Fix optional params not applying to all interfaces (#20313)

Currently optional params only get applied to the first child port in group. We want optional parameters to apply to all child ports in group.

For example:

        "Ethernet0": {
            "default_brkout_mode": "2x400G",
            "autoneg": "on"
        },
In this scenario we want autoneg on to be in the config_db entry for
both interfaces that belong to the group; Ethernet0 and Ethernet4.

Currently this only gets applied to Ethernet0:

(Pdb++) pp child_ports
{'Ethernet0': {'alias': 'Ethernet1/1',
               'autoneg': 'on',
               'index': '1',
               'lanes': '17,18,19,20',
               'speed': '400000',
               'subport': '1'},
 'Ethernet4': {'alias': 'Ethernet1/5',
               'index': '1',
               'lanes': '21,22,23,24',
               'speed': '400000',
               'subport': '2'}}
With this change it now gets applied to all interfaces in the group as expected

(Pdb) pp child_ports
{'Ethernet0': {'alias': 'Ethernet1/1',
               'autoneg': 'on',
               'index': '1',
               'lanes': '17,18,19,20',
               'speed': '400000',
               'subport': '1'},
 'Ethernet4': {'alias': 'Ethernet1/5',
               'autoneg': 'on',
               'index': '1',
               'lanes': '21,22,23,24',
               'speed': '400000',
               'subport': '2'}}
Why I did it
I need to apply optional hwsku params for a group of interfaces.

How I did it
Just changed to iterate over every child port in group.

How to verify it
Introduction to the problem shows verification of the solution. Check the output manually and also fixed the unit test to correspond and pass.

arista-nwolfe committed:
Set `xoff` of `x86_64-arista_7800r3_48cq2_lc` to the same as`x86_64-arista_7800r3_48cqm2_lc` (#20926)

* Set xoff of x86_64-arista_7800r3_48cq2_lc to the same as
x86_64-arista_7800r3_48cqm2_lc

* Updating CW2 sample_output for unit tests

* Updating the py3 sample_output file as well

mssonicbld committed:
[submodule] Update submodule sonic-platform-daemons to the latest HEAD automatically (#21043)

#### Why I did it
src/sonic-platform-daemons
```
* 1962fca - (HEAD -> master, origin/master, origin/HEAD) Fixed bug in chassisd causing incorrect number of ASICs in CHASSIS_STATE_DB (#560) (4 hours ago) [Vivek Verma]
```
#### How I did it
#### How to verify it
#### Description for the changelog

gpunathilell committed:
[Smartswitch][Mellanox] Enable thermal updater class for DPUs on switch (#20945)

- Why I did it
In order to use the thermal data from DPUs to be used by hw-mgmt algorithms for controlling the speed of the fans.
Changes in

thermal_manager.py : Updated condition to consider if DPUs are present along with host management mode
thermal_updater.py: Added new APIs for DPUs and made configuration to update start and stop based on the initialization

- How I did it
Parse the DPUs information from the chassis_instance modules, get the temperature information from the module.get_temperature_dict information, This will be added to the Platform API PR

- How i tested it
Added new unit test to cover the thermal algo.

vivekrnv committed:
[SmartSwitch] Skip xcvrd on the DPU (#20960)

- Why I did it
Data port on the DPU is of type "Dpc" i..e no xcvr is present on it. Thus xcvrd can effectively be skipped on this platform

- How I did it
change the pmon configuration
"skip_ycabled": true,
"skip_xcvrd": true

- How to verify it
Manual test to ensure xcrvd is skipped for this type of ports.
Load and check if xcvrd is not started

dgsudharsan committed:
[FRR]Setting multipath size to 514 and disabling bgp-vnc for optimization (#20744)

- Why I did it
Increased the max multipath to 512. In addition removed bgp-vnc as this feature is not used by SONiC .
https://docs.frrouting.org/en/latest/vnc.html#vnc-and-vnc-gw
This feature adds overhead in general and removing this improves bgp convergence time for scale.

- How I did it
Added a patch modify debian rules.

- How to verify it
Run BGP tests in scaled topology.

dgsudharsan committed:
[BGP]Increasing maximum-paths to 514 (#20745)

- Why I did it
This PR follows #20744
Increasing the maximum-paths to 514. Currently value of 64 restricts the maximum paths that can created by BGP.

- How I did it
Updated constants file.

- How to verify it
Modified UT to verify.

mssonicbld committed:
[submodule] Update submodule sonic-dbsyncd to the latest HEAD automatically (#21040)

#### Why I did it
src/sonic-dbsyncd
```
* 7ad34f6 - (HEAD -> master, origin/master, origin/HEAD) [ci] Fix pipeline issue caused by dist upgrade (#72) (53 minutes ago) [Liu Shilong]
```
#### How I did it
#### How to verify it
#### Description for the changelog

jhli-cisco committed:
add express-boot support (#17369)

Add express-boot support

guoguojia2021 committed:
[FRR] Enable pathd in frr (#20683)

[FRR] Enable pathd in frr 

Signed-off-by: guozhongfeng.gzf@alibaba-inc.com

ashwnsri committed:
Code optimizations to reduce disk writes on SONiC switches (#20640)

Why I did it
This PR contains various optimizations to reduce disk writes on SONiC switches as mentioned in the High-level Design.

This contributes to the overall reduction in data written to the disk over time, which can help extend the life of the disk and reduce the risk of disk failure.

Work item tracking
Microsoft ADO (number only): 28300356

How I did it
Moved supervisord logs within swss to tmpfs
Optimized away repeated disk writes by vtysh in the bgp container
Moved monit and logrotate writers to tmpfs

How to verify it
Flash image with these optimizations onto a sonic switch and check the following:

Within swss container, supervisord logs are written to /dev/shm/supervisor/
Within bgp container, verify that ~/.history_frr file is not written into repeatedly with show bgp summary json
Verify that monit state file is in /dev/shm/monit
Verify that logrotate status file is in /dev/shm/logrotate

gpunathilell committed:
[Mellanox][Smartswitch] Platform API for Smartswtich (#20944)

[Mellanox][Smartswitch] Platform API for Smartswtich

wen587 committed:
Disable change on minigraph and init_cfg.json (#19597)

Why I did it
The further change of minigraph.py and init_cfg.json should be blocked to align with PROD env.

How I did it
Add checker to block change

How to verify it
PR test

bradh352 committed:
[yang] LOGGER missing require_manual_refresh (#20969)

Why I did it
YANG failures during config replace with default configuration.

As of #19611 and sonic-net/sonic-utilities#3428 a new database field of require_manual_refresh was added.

This leads to YANG failures in the factory-default configuration for the xcvrd log entry which by default is written as true.

Work item tracking
How I did it
Updated YANG file for new field.

How to verify it
Verify config replace works with factory default configuration.

anamehra committed:
Update cisco-8000.ini to 202405.0.13 release (#21051)

Release Notes:

Fixed MIGSMSFT-752: missing LC data from "show chassis module status" CLI output
Fixed MIGSMSFT-504: FC serial number update issue after FC swap/change
Fixed MIGSMSFT-780: fix the issue: show_techsupport & saidump errors. Included rib-cli in syncd docker.
Fixed MIGSMSFT-761: cardevent.py error due to connection reset
Fixed syncd grpcpp srv crash

Signed-off-by: Anand Mehra anamehra@cisco.com

Gfrom2016 committed:
Upgrade xgs SAI version tio 11.2.16.2 (#21068)

Why I did it
Upgrade SAI xgs version to 11.2.16.2 to fix #20725

Work item tracking
Microsoft ADO (number only): 30445459

How I did it
Update xgs SAI version to 11.2.16.2 in the sai.mk file.

How to verify it
load the built image to the DUT, swss starts fine.

Signed-off-by: zitingguo-ms <zitingguo@microsoft.com>

mssonicbld committed:
[submodule] Update submodule sonic-sairedis to the latest HEAD automatically (#21066)

src/sonic-sairedis

* 4f7eed0a - (HEAD -> master, origin/master, origin/HEAD) Revert back to SAI version 1 15 (#1481) (80 minutes ago) [Ying X

dprital committed:
[Mellanox] Change NVIDIA SDK build to cmake method (#19810)

- Why I did it

Up to now, building Mellanox SDK debian packages was done by "autotools" method.
Lately, Mellanox SDK team added the support to build by "cmake" method.
The benefit of using "cmake" is with decrease time of SDK build as "cmake" make it faster then "auto build".

When moving to "cmake" method, all SDK debians:

applibs
sx-complib
sx-example
sx-gen-utils
sx-scew
sxd-libs
python-sdk-api
sx-acl-helper
wjh-libs
sx-hash-calc
sx-obj-desc-lib
sxd-libs
will be replaced by single debian - sys_sdk.

- How I did it

Update cmake version on sonic slave dockers (bookworm and bullseye)
Change SDK Makefiles, delete all Makefiles of removed debians and add new Makefile for building sys_sdk.
Change sdk.mk and sdk.dep acordingly.
sx-kernel package was remain since it is consumed by the host as well and I didn't want to consume sys_sdk in host.

- How to verify it

Enter syncd docker and verify the list of SDK packages are sx-kernel and sys-sdk only
Run regression tests and verify there is no degradation
Install the debian that include dbg-symbols and verify there are debug symbols by setting relevant breakpoints
Test the build time when using "cmake" for building SDK packages, compare to "auto build"

peterbailey-arista committed:
Enable SFF Manger on Arista linecards by default (#20886)

SFF Manager in xcvrd needs to run to take non-cmis transceivers out of
lpmode as part of the fix for these transceivers remaining in down state.

stephenxs committed:
Support bulk chunk size in bulk counter polling operation (#20754)

Why I did it
Added a new field in FLEX_COUNTER_TABLE to represent the bulk chunk size for bulk counter polling.

mssonicbld committed:
[submodule] Update submodule sonic-platform-daemons to the latest HEAD automatically (#21084)

#### Why I did it
src/sonic-platform-daemons
```
* 60e7224 - (HEAD -> master, origin/test_202411_20, origin/test_202411_100, origin/master, origin/kperumal_202411, origin/HEAD, origin/202411) thermalctld: Add support for fans on non-CPU modules (#555) (9 hours ago) [Patrick MacArthur]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-dbsyncd to the latest HEAD automatically (#21065)

#### Why I did it
src/sonic-dbsyncd
```
* b0ea01f - (HEAD -> master, origin/test_202411_20, origin/test_202411_100, origin/master, origin/kperumal_202411, origin/HEAD, origin/202411) [lldp]Fix the issue of only one field lldp_rem_time_mark in APPL_DB (#71) (2 days ago) [Zhaohui Sun]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-sairedis to the latest HEAD automatically (#21085)

#### Why I did it
src/sonic-sairedis
```
* 92322be7 - (HEAD -> master, origin/master, origin/kperumal_202411, origin/HEAD, origin/202411) Update the Azure Pipeline build to use Bookworm (#1452) (29 hours ago) [Saikrishna Arcot]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-swss-common to the latest HEAD automatically (#21014)

#### Why I did it
src/sonic-swss-common
```
* 7425c42 - (HEAD -> master, origin/master, origin/HEAD, origin/202411) Update the Azure Pipeline build to use Bookworm and Ubuntu 22.04 (#937) (26 hours ago) [Saikrishna Arcot]
* fb6ce44 - (origin/test_202411_20, origin/test_202411_100, origin/kperumal_202411) Add definition for MEMORY_STATISTICS table in schema (#898) (4 days ago) [Arham-Nasir]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-linux-kernel to the latest HEAD automatically (#21011)

#### Why I did it
src/sonic-linux-kernel
```
* c281fc9 - (HEAD -> master, origin/test_202411_20, origin/test_202411_100, origin/master, origin/kperumal_202411, origin/HEAD, origin/202411) Integrate HW-MGMT 7.0040.2000 Changes (#446) (4 days ago) [DavidZagury]
```
#### How I did it
#### How to verify it
#### Description for the changelog

skr31 committed:
[Mellanox] removed component_versions.dep file (#19982)

- Why I did it
The component versions file that is created during compilation was not being updated because it was being cached, even when some of the versions on it should have been updated.

- How I did it
Removed component-versions.dep

opcoder0 committed:
Build and include tacacs+ for PTF Py3 only image (#20701)

Why I did it
The "tacacs+" package is not available in the apt repository for Debian Bullseye and beyond. The sonic-mgmt tests require tacacs+ and this PR builds it from source and adds it to the PY3 only image.

Work item tracking
Microsoft ADO (number only): 29946202

How I did it
Clone tacacs+ from https://github.com/robot527/tac_plus build and install it.

How to verify it
By running sonic-mgmt restart-ptf step with newly built image.

Tested branch (Please provide the tested image version)
Not applicable.

Description for the changelog
[docker-ptf]: Build and include tacacs+ for py3only image

The "tacacs+" package is not available in the apt repository for Debian Bullseye and beyond. The sonic-mgmt tests require tacacs+ and this commit builds it from source and adds it to the PY3 only image.
Link to config_db schema for YANG module changes
Not applicable.

mssonicbld committed:
[submodule] Update submodule sonic-gnmi to the latest HEAD automatically (#21105)

#### Why I did it
src/sonic-gnmi
```
* 6eade51 - (HEAD -> master, origin/master, origin/HEAD) Merge pull request #319 from hdwhdw/upgrade (7 hours ago) [Dawei Huang]
* 5d68f51 - Merge branch 'sonic-net:master' into upgrade (2 weeks ago) [Dawei Huang]
* d62df54 - format fix. (5 weeks ago) [Dawei Huang]
* 874c6ec - Merge branch 'master' into upgrade (5 weeks ago) [Dawei Huang]
* 2a11f04 - DBUS client code and unit test. (5 weeks ago) [Dawei Huang]
* 2846d28 - Add dbus image counter (5 weeks ago) [Dawei Huang]
```
#### How I did it
#### How to verify it
#### Description for the changelog

mssonicbld committed:
[submodule] Update submodule sonic-platform-common to the latest HEAD automatically (#21108)

#### Why I did it
src/sonic-platform-common
```
* e955e58 - (HEAD -> master, origin/master, origin/HEAD) get_error_description should return 'OK' instead of None when there is no error (#518) (18 hours ago) [Kebo Liu]
```
#### How I did it
#### How to verify it
#### Description for the changelog

wajahatrazi committed:
Updated YANG model for PVST & MST including testcases

wajahatrazi committed:
Updated YANG model for PVST & MST including testcases
